src/exposition/http.rs

```diff
@@ -256,3 +347,63 @@ mod handlers {
         }
     }
 }
+
+#[derive(Serialize, Deserialize)]
+pub struct Metadata {
+    counters: Vec<MetricMetadata>,
+    gauges: Vec<MetricMetadata>,
+    histograms: Vec<(MetricMetadata, histogram::Config)>,
```

Comment:
Not a big deal, but the snapshot also contains the config, so this data is being duplicated.

Reply:
I think the metadata should contain all the information to build the schema. If anything, I would drop the config from the snapshot, but that's not possible with the current interfaces to `histogram::Snapshot`.

Reply:
Seems reasonable.